src/read.c

```diff
@@ -1441,7 +1441,7 @@ static avifResult avifDecoderGenerateImageGridTiles(avifDecoder * decoder, avifI
 
     if (tilesAvailable != grid->rows * grid->columns) {
         avifDiagnosticsPrintf(&decoder->diag,
-                              "Grid image of dimensions %ux%u requires %u tiles, and only %u were found",
+                              "Grid image of dimensions %ux%u requires %u tiles, but %u were found",
```

Comment:
Is the check here redundant now that we validate `grid->rows * grid->columns` in `avifDecoderItemReadAndParse`?

Reply:
i thought about this as well, the only subtle difference is that this check happens for all grid items (not just those found in the actual input).

so there is no harm in keeping this. or if you prefer i can change this one into an assert and document why that is sufficient.